cucumber-junit-platform-engine/src/main/java/io/cucumber/junit/platform/engine/FeatureElementOrderingVisitor.java

```diff
@@ -0,0 +1,23 @@
+package io.cucumber.junit.platform.engine;
+
+import org.junit.platform.engine.TestDescriptor;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Comparator.comparing;
+
+final class FeatureElementOrderingVisitor implements TestDescriptor.Visitor {
+
+    @Override
+    public void visit(TestDescriptor descriptor) {
+        // TODO: Make order configurable
+        List<? extends TestDescriptor> sortedChildren = descriptor.getChildren().stream()
+                .filter(FeatureElementDescriptor.class::isInstance)
+                .map(FeatureElementDescriptor.class::cast)
+                .sorted(comparing(FeatureElementDescriptor::getLine))
+                .collect(Collectors.toList());
+        sortedChildren.forEach(descriptor::removeChild);
+        sortedChildren.forEach(descriptor::addChild);
```

Comment:
I wonder if we should introduce an `TestDescriptor.orderChildren(UnaryOperator<TestDescriptor> orderer)` method (or similar). We could also make use of that in Jupiter. :thinking:

Reply:
That would be most useful! Though I'm not sure I understand the signature. Were you thinking of `Comparator<TestDescriptor>`?

Reply:
Sorry, I meant `UnaryOperator<List<TestDescriptor>>`. While a `Comparator` makes sense as well, in some cases (think `Collections.shuffle`) it's easier to return a new list. The implementation should verify that no elements were added or removed.